services/brig/src/Brig/User/API/Auth.hs

```diff
@@ -233,6 +235,14 @@ reAuthUserH (uid ::: req) = do
 reAuthUser :: UserId -> ReAuthUser -> (Handler r) ()
 reAuthUser uid body = do
   User.reauthenticate uid (reAuthPassword body) !>> reauthError
+  case reAuthCodeAction body of
+    Just action ->
+      Auth.verifyCode (reAuthCode body) action uid
+        `catchE` \case
+          VerificationCodeNoPendingCode -> throwE $ reauthError ReAuthNoPendingCode
+          VerificationCodeRequired -> throwE $ reauthError ReAuthVerificationCodeRequired
+          VerificationCodeNoEmail -> throwE $ reauthError ReAuthNoEmail
```

Comment:
Can't we just throw what we get from `Auth.verifyCode`, instead of introducing new error types?  (Maybe we can't, but why?)

Reply:
Ah, maybe because the json doesn't fit?  (Should have checked myself...)

Reply:
`verifyCode` throws certain errors. But this can be called in different places and is mapped to the error types that are required from where it is called. In this case we map to `ReAuthError`. Not sure right now if this can be done better.